### DISCUSSION TITLE
What feature would you like to see next?

### DISCUSSION THREAD

**Discussion by shannonhochkins** [shannonhochkins/ha-component-kit#28] [Polls]
I have a bunch of ideas on what to include next, I would appreciate some help in what to tackle next!

Or suggest something completely different in the comments :)

**Comment by hillaliy** [comment]
Hi friend, I ❤️your project.
I'm a developer myself.
I'll try to add VacuumCard and maybe I can help with other things.

**Reply by shannonhochkins** [reply]
That would be great if you'd like to contribute!

**Reply by hillaliy** [reply]
Hi it's not easy to dive to the code. I start to add the VacuumCard and have some problem. If you have some time I'll appreciate your help. Maybe we can chat on Discord. If you want I can publish the branch.

**Reply by shannonhochkins** [reply]
Hey @hillaliy, feel free to message me here I'm pretty responsive! I hadn't thought to open up something on discord, but if you like, you can ping me on discord! username is `_jinglez_`

**Comment by juan0x00** [comment]
Hello, I have tried this project, and it's impressive, great job!!!. 
I just voted for the second option because I would like to be able to add entity cards like sensors, for example, a temperature or humidity sensor. Lastly, could we add some time graphs for these sensors?

**Reply by shannonhochkins** [reply]
Thanks jaun! Awesome! Sensor cards aren't that difficult so that could be a quick addition! All of your ideas are on my list ;) just curious, did you install the packages manually or use the npm create command?

**Reply by juan0x00** [reply]
I would like to say the same, but I'm not as good at programming as you are!! Thank you for keeping my ideas in mind!! I followed your video tutorial, so did I install it manually? not sure

**Reply by shannonhochkins** [reply]
That's manually, but that's okay! I've only released the create command recently and haven't made a video about it yet!

Thank you for your compliments :)

**Comment by pdhruv93** [comment]
Hi. When can we expect the Calendar Card?

**Reply by shannonhochkins** [reply]
I'm currently focusing on a home assistant addon, you're welcome to design something and send it over as a potential integration, I'm doing a lot of work by myself, sorry it's hard to achieve everything

**Comment by ifuller1** [comment]
@shannonhochkins - it doesn't look like callService is able to handle a response? I'm hoping to use weather forecasts but that's now handled with weather.get_forecast. Is this something you might update?

Cheers

**Reply by shannonhochkins** [reply]
Unfortunately, callService is simply a trigger, if you want to handle responses you can use connection.sendMessagePromise and send through the required parameters

```
const { useStore } = useHass();
const connection = useStore();
```

**Reply by shannonhochkins** [reply]
To determine the parameters, you're best inspecting the websockets sent from home assistant when you trigger get forecast, then copy over the arguments to sendMessagePromise

**Reply by shannonhochkins** [reply]
You should already have the forecasts value on a weather entity under the attributes or is this not what you're after?

**Comment by ifuller1** [comment]
That's going away in home assistant.

https://www.home-assistant.io/blog/2023/09/06/release-20239/#:~:text=With%20this%20change%2C%20the%20forecast,being%20updated%20all%20the%20time
.

**Reply by shannonhochkins** [reply]
Thanks for the heads up! I'll migrate this, looks like it won't be removed until the next minor version so we have time!

I'll ensure it's all fixed up, I've been wanting to add toggles to swap between forecast views always

I'm almost finished with the calendar card then I'll jump on this 🤠

**Reply by shannonhochkins** [reply]
Just a heads up @ifuller1 - the latest release now uses the newer web socket subscription method rather than the the forecast attribute for the weather card :) 

If you want to use the forecasts outside of the card, you can use the new `useWeather` hook :)

**Comment by ifuller1** [comment]
Thanks. Will this add response values for all service calls?

**Reply by shannonhochkins** [reply]
Negative, the api proxy object Is a proxy foe the callService function from home assistant, unfortunately there's a few methods like getForecast that indicate that you will receive a response, but this is a problem with the home assistant twebsocket package not my package

**Reply by shannonhochkins** [reply]
However, I am also working on a new function to make api requests which I've implemented as part of the calendar card, which opens up doors to a hell of a lot more customisation like creating or updating properties of entities inside home assistant 😀

**Comment by yann510** [comment]
I was looking for a better way than fighting with yaml configs to create custom experiences and found this.
It looks fantastic, and I can't wait to start a new custom dashboard using React and your set of components :)

One thing I would like is the ability for the speaker card to have multiple speakers and to group/ungroup the speakers.
Do you have any plans to add these features to the speaker card?

**Reply by shannonhochkins** [reply]
Thanks mate! It should already support the functionality, is it not working for you? Have a look at the docs it supports speaker group members but it is untested as I don't have speakers that support it!

**Reply by yann510** [reply]
Oh I'm sorry I didn't know it was already supported!!
I'll test it out soon and create an issue + pr if anything needs to be fixed :)

**Reply by shannonhochkins** [reply]
I just checked this a little more, if the media_player does support grouping, it should show an additional button on the card, there's a   groupMembers prop available on the mediaPlayer, if your entity supports grouping, it will render the card, if it doesn't, it will throw an error

However, there's other solutions i didn't think of, in Google Home for example, you can create a speaker group, which creates one media_player entity in home assistant, but linked to all speakers within the group, home assistant still thinks the media player itself doesn't support grouping, so I may provide another way around this to show volume / controls for related entities

**Reply by yann510** [reply]
You should get 2 speakers which you can group just to play with this 😂
I'm kidding obviously, but I'll take a look once I have some free time during the upcoming week.

**Reply by shannonhochkins** [reply]
Haha, no dramas bro

**Comment by eivindholvik** [comment]
Would there be a way of opening a modal when the door bell rings with the live feed? A couple of buttons to respond, open the door, or just ignore (close the modal)

If there is not, is that something that could be implemented in the future?

This looks great by the way! I'm just getting started today, so nothing is up yet. I am still getting errors for using http://xxx:8123 as VITE_HA_URL for some reason :D

Cannot connect to Home Assistant instances over "http://".

**Reply by jenseo** [reply]
About the https thing, make sure to set _allowNonSecure_ on _HassConnect_

**Reply by shannonhochkins** [reply]
I'll change that error message to include those instructions as well. As for your idea about the modal, yes, that's absolutely possible I do it on my dashboard, I will send over an example when I'm on my computer next

**Comment by shannonhochkins** [comment]
You could do something like this @eivindholvik 

```
import { useState, useId  } from 'react';
import { Modal, CameraCard  } from '@hakit/components';
import { useEntity } from '@hakit/core';

export function FrontDoorRing() {
  // change to your entity name
  const camera = useEntity('sensor.ring_camera');
  const _id = useId();
  const [open, setOpen] = useState(false);
  // the state value might be different to what you're expecting
  const cameraActive = camera.state === 'on';
  if (cameraActive && !open) {
    setOpen(true);
  } else if (!cameraActive && open) {
    setOpen(false);
  }
  return <Modal title="Someones at the door!" open={open} onClose={() => {
    setOpen(false);
  }}>
    // insert picture from the camera, or a live feed using the CameraCard or sub components
    <CameraCard entity="camera.ring_doorbell" view="live"  />
  </Modal>
}
```
